go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -864,10 +863,7 @@ func (sess *Session) AddDB(ctx *sql.Context, dbState InitialDbState) error {
 		sessionState.Err = dbState.Err
 	} else if dbState.WorkingSet != nil {
 		sessionState.WorkingSet = dbState.WorkingSet
-		workingRoot := dbState.WorkingSet.WorkingRoot()
-		// logrus.Tracef("working root intialized to %s", workingRoot.DebugString(ctx, false))
-
-		err := sess.setRoot(ctx, db.Name(), workingRoot)
+		err := sess.setWorkingSet(ctx, db.Name(), dbState.WorkingSet)
```

Comment:
Seems like you want to remove the line above, and move the consistency check into the exported version of this method.